File: server/src/main/scala/producer/message_generator/compression_type/CompressionType.scala
package producer.message_generator.compression_type

import com.tools.teal.pulsar.ui.producer.v1.producer as pb
import org.apache.pulsar.client.api.CompressionType as PulsarCompressionType

type CompressionType = PulsarCompressionType

object CompressionType:
    def fromPb(v: pb.CompressionType): PulsarCompressionType  =
        v match
            case pb.CompressionType.COMPRESSION_TYPE_LZ4 => PulsarCompressionType.LZ4
            case pb.CompressionType.COMPRESSION_TYPE_ZLIB => PulsarCompressionType.ZLIB
            case pb.CompressionType.COMPRESSION_TYPE_ZSTD => PulsarCompressionType.ZSTD
            case pb.CompressionType.COMPRESSION_TYPE_NONE => PulsarCompressionType.NONE
            case pb.CompressionType.COMPRESSION_TYPE_SNAPPY => PulsarCompressionType.SNAPPY
            case _ => throw Exception(s"Unknown compression type: $v")


    def toPb(v: PulsarCompressionType): pb.CompressionType =
        v match
            case PulsarCompressionType.LZ4 => pb.CompressionType.COMPRESSION_TYPE_LZ4
            case PulsarCompressionType.ZLIB => pb.CompressionType.COMPRESSION_TYPE_ZLIB
            case PulsarCompressionType.ZSTD => pb.CompressionType.COMPRESSION_TYPE_ZSTD
            case PulsarCompressionType.NONE => pb.CompressionType.COMPRESSION_TYPE_NONE
            case PulsarCompressionType.SNAPPY => pb.CompressionType.COMPRESSION_TYPE_SNAPPY
            case _ => throw Exception(s"Unknown compression type: $v")

File: ui/components/ui/ProducerSession/producer-task/pulsar-producer-config/compression-type/compression-type.ts
import * as pb from '../../../../../../grpc-web/tools/teal/pulsar/ui/producer/v1/producer_pb';

export type CompressionType = 'none' | 'lz4' | 'zlib' | 'zstd' | 'snappy';

export function compressionTypeFromPb(compressionType: pb.CompressionType): CompressionType {
  switch (compressionType) {
    case pb.CompressionType.COMPRESSION_TYPE_NONE: return 'none';
    case pb.CompressionType.COMPRESSION_TYPE_LZ4: return 'lz4';
    case pb.CompressionType.COMPRESSION_TYPE_ZLIB: return 'zlib';
    case pb.CompressionType.COMPRESSION_TYPE_ZSTD: return 'zstd';
    case pb.CompressionType.COMPRESSION_TYPE_SNAPPY: return 'snappy';
    default: throw new Error(`Unknown compression type: ${compressionType}`);
  }
}

export function compressionTypeToPb(compressionType: CompressionType): pb.CompressionType {
  switch (compressionType) {
    case 'none': return pb.CompressionType.COMPRESSION_TYPE_NONE;
    case 'lz4': return pb.CompressionType.COMPRESSION_TYPE_LZ4;
    case 'zlib': return pb.CompressionType.COMPRESSION_TYPE_ZLIB;
    case 'zstd': return pb.CompressionType.COMPRESSION_TYPE_ZSTD;
    case 'snappy': return pb.CompressionType.COMPRESSION_TYPE_SNAPPY;
  }
}

